src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -459,8 +459,43 @@ def verify_property(instance, condition):
 index_or_filter_regex = re.compile(r'\[(.*)\]')
 
 
+def _split_key_value_pair(expression):
+
+    def _find_split():
+        """ Find the first = sign to split on (that isn't in [brackets])"""
+        key = []
+        value = []
+        brackets = False
+        chars = list(expression)
+        while len(chars) > 0:
+            c = chars.pop(0)
+            if c == '=' and not brackets:
+                # keys done the rest is value
+                value = chars
+                break
+            elif c == '[':
+                brackets = True
+                key += c
+            elif c == ']' and brackets:
+                brackets = False
```

Comment:
I assume the `[` can never be nested like  `foo[bar[p]==0]=v1`? 
If nesting can never happen, you probably don't need to have a loop, rather check indices of `=`, and `[`, and see which comes first, and then split based on that. But looping is also fine by me
If nesting can happen, then you need a stack to match the pair, and I hope it is not the case.